### ISSUE TITLE
Add an option to read json from file

### ISSUE THREAD

**Issue by the42** [twpayne/go-jsonstruct#9]
rationale: I would like to use go-jsonstruct in a `//go:generate`-directive. However the [documentation](https://pkg.go.dev/cmd/go/internal/generate) says:

    The arguments to the directive are space-separated tokens or double-quoted strings passed to the generator as individual arguments when it is run.

So constructs like `gojsonstruct < file.json` or `cat file.json | gojsonstruct` will not work. Instead I have to use

    //go:generate sh -c "gojsonstruct -packagename $GOPACKAGE < ./file.json"

Which is not portable. If gojsonstruct would be able to read from file the above generator statement could be made environment neutral.

**Comment by the42** [comment]
and for the same reason write the output to a file

**Comment by twpayne** [comment]
This is now done. You can pass files to be read as arguments to `gojsonstruct` and set the output file with the `-o` flag.

**Comment by the42** [comment]
I see that with commit #10 reding from a file and #061ce2f595c4bb7d0f98f1cf4283b3d08b79fcf6 output with -o should have been added.

Yet after `go install github.com/twpayne/go-jsonstruct/cmd/gojsonstruct@latest` I do not have this functionality.

I tried gojsonstruct -format json <filename> as I interpret this to be the workings of ObserveJSONFile yet the content isn't read from the file.

Also the -o <filenanme> is also not respected.

I also see that you introduced  a v2 - tag. Maybe go install @latest will not work?  I tried `go install github.com/twpayne/go-jsonstruct/cmd/gojsonstruct@v2` and `go install github.com/twpayne/go-jsonstruct/cmd/gojsonstruct@v2.0.0` yet both commands fail.

Maybe you have a local go.mod or go.work with replaces?
